zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/StreamProcessor.java

```diff
@@ -372,30 +371,35 @@ private void onRecovered(final LastProcessingPositions lastProcessingPositions)
     streamProcessorContext.streamProcessorPhase(Phase.PROCESSING);
     metrics.setStreamProcessorProcessing();
 
-    AsyncUtil.chainSteps(
-        0,
-        new Step[] {
-          () -> processorActorService.open(actor), () -> asyncScheduleServiceContext.submitActors()
-        },
-        () -> startProcessing(lastProcessingPositions),
-        this::onFailure);
+    processorActorService
+        .open(actor)
+        .andThen(() -> asyncScheduleServiceContext.submitActors(actor), actor)
+        .onComplete(
+            (ignored, error) -> {
+              if (error != null) {
+                onFailure(error);
+              }
+              startProcessing(lastProcessingPositions);
```

Comment:
:question: should we start processing even  if there is an error in the future?

Reply:
Oops, no we shouldn't :grimacing: 
I think that the `onFailure` will prevent that anyway but let's make it clearer :+1: